core/services/keeper/registry_synchronizer_process_logs.go

```diff
@@ -174,9 +175,9 @@ func (rs *RegistrySynchronizer) handleUpkeepPerformed(broadcast log.Broadcast) {
 		rs.logger.AssumptionViolationf("expected UpkeepPerformed log but got %T", log)
 		return
 	}
-
-	// set last run to 0 so that keeper can resume checkUpkeep()
-	err = rs.orm.SetLastRunHeightForUpkeepOnJob(rs.job.ID, log.Id.Int64(), 0)
```

Comment:
This was being set to 0 before which is used in the select query "upkeep_registrations.last_run_block_height = 0". I think if it's not set to 0 then the same keeper will not process another upkeep for gracePeriod blocks which will break stuff. 
(It's very convoluted and I think we should fix the gracePeriod logic also)

Reply:
no longer doing this. I think it now is a little simpler imo. I will draw out a flow diagram to add to the doc

Reply:
no longer doing this. I think it now is a little simpler imo. I will draw out a flow diagram to add to the doc

Reply:
nice work :clap: